org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/json/JsonEventProcessor.java

```diff
@@ -32,12 +34,25 @@ public JsonEventProcessor(IUiSession uiSession) {
   public void processEvents(final JsonRequest request, final JsonResponse response) {
     Assertions.assertTrue(ModelJobs.isModelThread(), "Event processing must be called from the model thread  [currentThread={}, request={}, response={}]",
         Thread.currentThread().getName(), request, response);
+
     for (final JsonEvent event : request.getEvents()) {
       processEvent(event, response);
     }
   }
 
   protected void processEvent(JsonEvent event, JsonResponse response) {
+    IJsonAdapter<?> jsonAdapter = m_uiSession.getJsonAdapter(event.getTarget());
+    addNameToContext(getName(jsonAdapter.getModel().getClass().getName(), event.getType()));
+
+    processEventInternal(event, response);
+  }
+
+  protected String getName(String fullName, String eventType) {
+    String adapterModel = fullName.substring(fullName.lastIndexOf('.') + 1);
+    return adapterModel + "." + eventType;
+  }
+
+  protected void processEventInternal(JsonEvent event, JsonResponse response) {
     final IJsonAdapter<?> jsonAdapter = m_uiSession.getJsonAdapter(event.getTarget());
     if (jsonAdapter == null) {
       LOG.info("No adapter found for event. {}", event.toSafeString());
```

Comment:
Try to prevent unnecessary memory/cpu allocation if tracing is deactivated:
```suggestion
  protected void processEvent(JsonEvent event, JsonResponse response) {
    final IJsonAdapter<?> jsonAdapter = m_uiSession.getJsonAdapter(event.getTarget());
    if (jsonAdapter == null) {
      LOG.info("No adapter found for event. {}", event.toSafeString());
      return;
    }
    try {
      if (LOG.isDebugEnabled()) {
        LOG.debug("Handling event '{}' for adapter with ID {}", event.getType(), event.getTarget());
      }

      addNameToContext(() -> buildSpanName(event, jsonAdapter));
      jsonAdapter.handleUiEvent(event);
```

```
  protected String buildSpanName(JsonEvent event, IJsonAdapter<?> jsonAdapter) {
    return jsonAdapter.getModel().getClass().getSimpleName() + "." + event.getType();
  }
```